src/Core/Framework/DataAbstractionLayer/Dbal/EntityReader.php

```diff
@@ -546,7 +546,11 @@ private function loadOneToManyWithPagination(
             $parentId = $entity->get('parentId');
 
             //extract mapping ids for the current entity
+            $mappingIds = [];
+			
+			if($parentId){
```

Comment:
If no parentId is found you could easily continue the foreach.
Please change it that way.

And like @shyim wrote: Add some tests for this case :blush: